// react-event-injector/dist/es5/propTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var prop_types_1 = require("prop-types");
exports.injectorTypes = {
    children: prop_types_1.oneOfType([prop_types_1.node, prop_types_1.func]),
    pure: prop_types_1.bool,
    settings: prop_types_1.shape({
        passive: prop_types_1.bool,
        capture: prop_types_1.bool
    })
};
exports.eventTypes = process.env.NODE_ENV === 'production'
    ? {}
    : {
        onCopy: prop_types_1.func,
        onCopyCapture: prop_types_1.func,
        onCut: prop_types_1.func,
        onCutCapture: prop_types_1.func,
        onPaste: prop_types_1.func,
        onPasteCapture: prop_types_1.func,
        onCompositionEnd: prop_types_1.func,
        onCompositionEndCapture: prop_types_1.func,
        onCompositionStart: prop_types_1.func,
        onCompositionStartCapture: prop_types_1.func,
        onCompositionUpdate: prop_types_1.func,
        onCompositionUpdateCapture: prop_types_1.func,
        onFocus: prop_types_1.func,
        onFocusCapture: prop_types_1.func,
        onBlur: prop_types_1.func,
        onBlurCapture: prop_types_1.func,
        onChange: prop_types_1.func,
        onChangeCapture: prop_types_1.func,
        onInput: prop_types_1.func,
        onInputCapture: prop_types_1.func,
        onReset: prop_types_1.func,
        onResetCapture: prop_types_1.func,
        onSubmit: prop_types_1.func,
        onSubmitCapture: prop_types_1.func,
        onInvalid: prop_types_1.func,
        onInvalidCapture: prop_types_1.func,
        onLoad: prop_types_1.func,
        onLoadCapture: prop_types_1.func,
        onError: prop_types_1.func,
        onErrorCapture: prop_types_1.func,
        onKeyDown: prop_types_1.func,
        onKeyDownCapture: prop_types_1.func,
        onKeyPress: prop_types_1.func,
        onKeyPressCapture: prop_types_1.func,
        onKeyUp: prop_types_1.func,
        onKeyUpCapture: prop_types_1.func,
        onAbort: prop_types_1.func,
        onAbortCapture: prop_types_1.func,
        onCanPlay: prop_types_1.func,
        onCanPlayCapture: prop_types_1.func,
        onCanPlayThrough: prop_types_1.func,
        onCanPlayThroughCapture: prop_types_1.func,
        onDurationChange: prop_types_1.func,
        onDurationChangeCapture: prop_types_1.func,
        onEmptied: prop_types_1.func,
        onEmptiedCapture: prop_types_1.func,
        onEncrypted: prop_types_1.func,
        onEncryptedCapture: prop_types_1.func,
        onEnded: prop_types_1.func,
        onEndedCapture: prop_types_1.func,
        onLoadedData: prop_types_1.func,
        onLoadedDataCapture: prop_types_1.func,
        onLoadedMetadata: prop_types_1.func,
        onLoadedMetadataCapture: prop_types_1.func,
        onLoadStart: prop_types_1.func,
        onLoadStartCapture: prop_types_1.func,
        onPause: prop_types_1.func,
        onPauseCapture: prop_types_1.func,
        onPlay: prop_types_1.func,
        onPlayCapture: prop_types_1.func,
        onPlaying: prop_types_1.func,
        onPlayingCapture: prop_types_1.func,
        onProgress: prop_types_1.func,
        onProgressCapture: prop_types_1.func,
        onRateChange: prop_types_1.func,
        onRateChangeCapture: prop_types_1.func,
        onSeeked: prop_types_1.func,
        onSeekedCapture: prop_types_1.func,
        onSeeking: prop_types_1.func,
        onSeekingCapture: prop_types_1.func,
        onStalled: prop_types_1.func,
        onStalledCapture: prop_types_1.func,
        onSuspend: prop_types_1.func,
        onSuspendCapture: prop_types_1.func,
        onTimeUpdate: prop_types_1.func,
        onTimeUpdateCapture: prop_types_1.func,
        onVolumeChange: prop_types_1.func,
        onVolumeChangeCapture: prop_types_1.func,
        onWaiting: prop_types_1.func,
        onWaitingCapture: prop_types_1.func,
        onClick: prop_types_1.func,
        onClickCapture: prop_types_1.func,
        onContextMenu: prop_types_1.func,
        onContextMenuCapture: prop_types_1.func,
        onDoubleClick: prop_types_1.func,
        onDoubleClickCapture: prop_types_1.func,
        onDrag: prop_types_1.func,
        onDragCapture: prop_types_1.func,
        onDragEnd: prop_types_1.func,
        onDragEndCapture: prop_types_1.func,
        onDragEnter: prop_types_1.func,
        onDragEnterCapture: prop_types_1.func,
        onDragExit: prop_types_1.func,
        onDragExitCapture: prop_types_1.func,
        onDragLeave: prop_types_1.func,
        onDragLeaveCapture: prop_types_1.func,
        onDragOver: prop_types_1.func,
        onDragOverCapture: prop_types_1.func,
        onDragStart: prop_types_1.func,
        onDragStartCapture: prop_types_1.func,
        onDrop: prop_types_1.func,
        onDropCapture: prop_types_1.func,
        onMouseDown: prop_types_1.func,
        onMouseDownCapture: prop_types_1.func,
        onMouseEnter: prop_types_1.func,
        onMouseLeave: prop_types_1.func,
        onMouseMove: prop_types_1.func,
        onMouseMoveCapture: prop_types_1.func,
        onMouseOut: prop_types_1.func,
        onMouseOutCapture: prop_types_1.func,
        onMouseOver: prop_types_1.func,
        onMouseOverCapture: prop_types_1.func,
        onMouseUp: prop_types_1.func,
        onMouseUpCapture: prop_types_1.func,
        onSelect: prop_types_1.func,
        onSelectCapture: prop_types_1.func,
        onTouchCancel: prop_types_1.func,
        onTouchCancelCapture: prop_types_1.func,
        onTouchEnd: prop_types_1.func,
        onTouchEndCapture: prop_types_1.func,
        onTouchMove: prop_types_1.func,
        onTouchMoveCapture: prop_types_1.func,
        onTouchStart: prop_types_1.func,
        onTouchStartCapture: prop_types_1.func,
        onPointerDown: prop_types_1.func,
        onPointerDownCapture: prop_types_1.func,
        onPointerMove: prop_types_1.func,
        onPointerMoveCapture: prop_types_1.func,
        onPointerUp: prop_types_1.func,
        onPointerUpCapture: prop_types_1.func,
        onPointerCancel: prop_types_1.func,
        onPointerCancelCapture: prop_types_1.func,
        onPointerEnter: prop_types_1.func,
        onPointerEnterCapture: prop_types_1.func,
        onPointerLeave: prop_types_1.func,
        onPointerLeaveCapture: prop_types_1.func,
        onPointerOver: prop_types_1.func,
        onPointerOverCapture: prop_types_1.func,
        onPointerOut: prop_types_1.func,
        onPointerOutCapture: prop_types_1.func,
        onGotPointerCapture: prop_types_1.func,
        onGotPointerCaptureCapture: prop_types_1.func,
        onLostPointerCapture: prop_types_1.func,
        onLostPointerCaptureCapture: prop_types_1.func,
        onScroll: prop_types_1.func,
        onScrollCapture: prop_types_1.func,
        onWheel: prop_types_1.func,
        onWheelCapture: prop_types_1.func,
        onAnimationStart: prop_types_1.func,
        onAnimationStartCapture: prop_types_1.func,
        onAnimationEnd: prop_types_1.func,
        onAnimationEndCapture: prop_types_1.func,
        onAnimationIteration: prop_types_1.func,
        onAnimationIterationCapture: prop_types_1.func,
        onTransitionEnd: prop_types_1.func,
        onTransitionEndCapture: prop_types_1.func,
    };
